extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -102,20 +102,35 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     try {
       const lookupResult = await this.view.pubLookup.lookupEmail(email);
       if (lookupResult && email) {
-        if (lookupResult.pubkey) {
-          const key = await KeyUtil.parse(lookupResult.pubkey);
-          if (!key.usableForEncryption && !KeyUtil.expired(key)) { // Not to skip expired keys
+        const pubkeys: Key[] = [];
+        for (const pubkey of lookupResult.pubkeys) {
+          const key = await KeyUtil.parse(pubkey);
+          if (!key.usableForEncryption && !key.revoked && !KeyUtil.expired(key)) { // Not to skip expired and revoked keys
             console.info('Dropping found+parsed key because getEncryptionKeyPacket===null', { for: email, fingerprint: key.id });
             Ui.toast(`Public Key retrieved for email ${email} with id ${key.id} was ignored because it's not usable for encryption.`, 5);
-            lookupResult.pubkey = null; // tslint:disable-line:no-null-keyword
+          } else {
+            pubkeys.push(key);
           }
         }
-        const ksContact = await ContactStore.obj({ email, name, pubkey: lookupResult.pubkey, lastCheck: Date.now() });
-        if (ksContact.pubkey) {
-          this.ksLookupsByEmail[email] = ksContact.pubkey;
+        // save multiple pubkeys as separate operations
+        // todo: add a convenient method to storage?
+        const updates: ContactUpdate[] = [];
+        if (!pubkeys.length) {
+          if (name) {
+            // update just name
+            updates.push({ name } as ContactUpdate);
+          } else {
+            return await ContactStore.obj({ email });
```

Comment:
Is this return here still to be addressed? It looks like returning here would cause none of the pubkeys to be saved below (this is probably why it's a draft - just checking).

Reply:
If true, than it would be great to find a way to test this, too. Currently tests are passing even without this object being stored.

Reply:
this code is inside
```
if (!pubkeys.length) {
```
so no pubkeys are available here.
I'll save the dummy object {email} as per your request.

Reply:
Got it. I think my original code was a bit of a spaghetti incident :) it's hard to follow which method has what responsibility and why, at least from reviewing the code. Your changes are good, I think it's my own code that confused me. A refactor of these methods in general may be nice, but that would be for another day.

Some of this complexity was for a reason (like responsiveness of UI as things are getting loaded from various, often remote places) and not all of these reasons were captured in tests, particularly UI responsiveness. Therefore refactoring this may not be easy, either.

Maybe a comment would help here - around that return:

> No public key found. Returning early, nothing to update in local store below.

Thanks!

Reply:
Done